tests/HttpServerTest.php

```diff
@@ -353,4 +353,194 @@ public function testTwoTypesOfEncodingForStreaming()
 
         $this->connection->emit('data', array($request));
     }
+
+    public function testNonNumericContentLengthResultsInError()
+    {
+        $callback = function(RequestInterface $request) {
+            $promise = new Promise(function ($resolve, $reject) use ($request) {
+                $request->getBody()->on('end', function() use ($resolve) {
+                    $resolve(new Response());
+                });
+            });
+                return $promise;
+        };
+
+
+        $request = "GET / HTTP/1.1\r\nHost: me.you\r\nContent-Length: bla\r\n\r\n";
+
+        $socket = new Socket($this->loop);
+        $server = new HttpServer($socket, $callback);
+
+        $socket->emit('connection', array($this->connection));
+
+        $this->connection->expects($this->once())->method('write')->with($this->equalTo("HTTP/1.1 400 Bad Request\r\n\r\n"));
+
+        $this->connection->emit('data', array($request));
+    }
+
+    public function testMultipleValuesInContentLengthResultsInError()
+    {
+        $callback = function(RequestInterface $request) {
+            $promise = new Promise(function ($resolve, $reject) use ($request) {
+                $request->getBody()->on('end', function() use ($resolve) {
+                    $resolve(new Response());
+                });
+            });
+                return $promise;
+        };
+
+
+        $request = "GET / HTTP/1.1\r\nHost: me.you\r\nContent-Length: 1\r\nContent-Length: 2\r\n\r\n";
+
+        $socket = new Socket($this->loop);
+        $server = new HttpServer($socket, $callback);
+
+        $socket->emit('connection', array($this->connection));
+
+        $this->connection->expects($this->once())->method('write')->with($this->equalTo("HTTP/1.1 400 Bad Request\r\n\r\n"));
+
+        $this->connection->emit('data', array($request));
+    }
+
+    public function testContenLenghtWithoutValueButBodyWillBeCutted()
+    {
+        $callback = function(RequestInterface $request) {
+            $promise = new Promise(function ($resolve, $reject) use ($request) {
+                $body = $request->getBody();
+                $content = '';
+
+                $body->on('data', function($data) use (&$content) {
```

Comment:
How often is this method invoked? How can you be sure? It looks like there's a subtle error here :-)